bundles/org.openhab.binding.opengarage/src/main/java/org/openhab/binding/opengarage/internal/OpenGarageConfiguration.java

```diff
@@ -12,14 +12,24 @@
  */
 package org.openhab.binding.opengarage.internal;
 
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+
 /**
  * The OpenGarageConfiguration class contains fields mapping thing configuration parameters.
  *
  * @author Paul Smedley - Initial contribution
  */
+@NonNullByDefault
 public class OpenGarageConfiguration {
-    public String hostname;
-    public long port = 80;
+    public @Nullable String hostname;
```

Comment:
`hostname` will never be null because you defined the parameter as required.
The best practice is to not defined it as `@Nullable` and set it to an empty string by default. In your thing handler (initialize), just check if its value is not empty.